### PR TITLE
[FIX JENKINS-37516] - js-extensions tests were broken

### PR DESCRIPTION
**Pull request by kzantow**
# Description

See [JENKINS-37516](https://issues.jenkins-ci.org/browse/JENKINS-37516).

Also got rid of the `.instance` for @tfennelly. Needs re-testing and publish before merge.
# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.
# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

File: js-extensions/index.js [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
+exports.store = new ExtensionStore();
 
-exports.Renderer = require('./dist/ExtensionRenderer.js').ExtensionRenderer;
+exports.Renderer = require('./dist/ExtensionRenderer.js').default;
 
-exports.classMetadataStore = require('./dist/ClassMetadataStore.js').instance;
+//in lieu of DI
+const ResourceLoadTracker = require('./dist/ResourceLoadTracker.js').default;
+const resourceLoadTracker = new ResourceLoadTracker();
+exports.store.ResourceLoadTracker = resourceLoadTracker;
```
Comment (tfennelly) [inline js-extensions/index.js]: So we `export` the `ExtensionStore` instance as `store` (lowercase) and we export the `ResourceLoadTracker` instance as `ResourceLoadTracker` (uppercase). Is that intentional? Why not both lowercase/uppercase?

Comment (tfennelly) [inline js-extensions/index.js]: And why `exports.store.ResourceLoadTracker` as opposed to just `exports.ResourceLoadTracker` ? Just curious. Maybe add a comment as to why it hangs there.

File: js-extensions/index.js [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
+
+//Put these in statics so we can mock them for testing. Ideally they would come from React scope.
+exports.Renderer.ExtensionStore = exports.store;
+exports.Renderer.ResourceLoadTracker = resourceLoadTracker;
```
Comment (tfennelly) [inline js-extensions/index.js]: Is there really a need for this? Seems to add unwanted noise in the top level API. For testing, can't you just do `require('./dist/ExtensionStore.js').default` and create a new instance from that ?

File: js-extensions/spec/ExtensionRenderer-spec.js [hunk 3]
```diff
@@ -76,7 +73,7 @@ describe('ExtensionRenderer', function () {
     });
 
     it('should do nothing interesting by default', function () {
-        const result = mount($(ExtensionRenderer, {extensionPoint: 'foo.bar.baz'}));
+        const result = mount(React.createElement(ExtensionRenderer, {extensionPoint: 'foo.bar.baz'}));
```
Comment (tfennelly) [inline js-extensions/spec/ExtensionRenderer-spec.js]: Ah yeah ... thanks for clearing that up !!

File: js-extensions/spec/ExtensionStore-spec.js [hunk 4]
```diff
@@ -1,7 +1,9 @@
 var jsTest = require('@jenkins-cd/js-test');
 var expect = require('chai').expect;
-var ExtensionStore = require('../dist/ExtensionStore').ExtensionStore;
-var ClassMetadataStore = require('../dist/ClassMetadataStore').instance;
+var ExtensionStore = require('../dist/ExtensionStore').default;
+var ClassMetadataStore = require('../dist/ClassMetadataStore').default;
+var resourceLoadTracker = new (require('../dist/ResourceLoadTracker').default)();
```
Comment (tfennelly) [inline js-extensions/spec/ExtensionStore-spec.js]: Ah there you go ... so do you really need the test related exports in the top level API? See earlier comment.

File: js-extensions/index.js (hunk 1 above)
Comment (kzantow) [inline js-extensions/index.js]: was trying to minimize code changes, might as well clean it up i guess. thanks for the cattle prod ;)

**Comment by michaelneale** [comment]
ping @tfennelly @kzantow can this be verified and merged?

**Comment by kzantow** [comment]
@tfennelly could you give this another look?

**Comment by tfennelly** [comment]
LGTM ... seems like you have some shrinkwrap head wrecking  to resolve.

**Comment by kzantow** [comment]
I need a 'works on my machine' badge for this one...

**Pull request merged**